public/js/app/components/SearchDepartments.jsx

```diff
@@ -36,10 +32,11 @@ function SearchDepartments({ onChange }) {
         <legend className="form-control-label">{departmentLabel}</legend>
         <select
           id="department"
-          defaultValue={initialDepartmentCode}
+          defaultValue={department}
```

Comment:
Hm, I think you should change this to value instead of defaultValue, but I not sure why it is even working without warning right now.

```suggestion
          value={department}
```

Reply:
I think I'm going to look into this a little more in detail and see what going on. I've a feeling that you could simplify things even more regarding the state and store.

Reply:
Thanks, I would be thankful to have your comments, I also have some ideas about optimizing the code even more when I am converting the js file to ts. I will update this PR with more optimization as well.

Reply:
I tested to remove a lot of our state (both from the store and from local component state) to only use the url (query-param) itself as source to the state and I looks promising I think. It simplifies things a bit at least.

I created a temp branch with my changes, so you can have a look at it if you want. And then we will look at it together next week. https://github.com/KTH/kopps-public-react-web/pull/271

Reply:
> I tested to remove a lot of our state (both from the store and from local component state) to only use the url (query-param) itself as source to the state and I looks promising I think. It simplifies things a bit at least.
> 
> I created a temp branch with my changes, so you can have a look at it if you want. And then we will look at it together next week. #271

I will check it for sure, thanks for bringing up this idea, it seems great ⭐️